streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractKeyValueStoreTest.java

```diff
@@ -58,6 +58,27 @@ public void after() {
         driver.clear();
     }
 
+    @Test
+    public void testDelete() {
+        store.close();
+        store = createKeyValueStore(driver.context(), Integer.class, String.class);
+
+        store.put(0, "zero");
+        store.put(1, "one");
+        store.put(2, "two");
+        store.delete(0);
+        store.put(1, null);
+        final KeyValueIterator<Integer, String> it = store.all();
+
+        while (it.hasNext()) {
+            final KeyValue<Integer, String> next = it.next();
+            // should not return deleted records in iterator
+            if (!next.key.equals(2)) {
+                fail("Got deleted key from iterator");
+            }
+        }
```

Comment:
The iterator should return exactly one record. This, we should add an `Assert.assertFalse(it.hasNext());` after the `if`

Reply:
Yes I used the new util just introduced in another PR.